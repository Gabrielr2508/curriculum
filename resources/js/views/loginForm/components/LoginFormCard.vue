<template>
	<v-flex xs12 sm6>
		<v-card>
			<v-layout align-center justify-center>
				<v-flex xs8 offset-xs-2>
					<v-img
						class="mt-4"
						src="../img/logo.png"
						contain
						width="100%"
						position="center"
					/>
				</v-flex>
			</v-layout>

			<v-card-text>
				<v-form
					v-model="valid"
					ref="loginForm"
				>
					<v-container>
						<v-layout column>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.email"
									:rules="rules.email"
									label="E-mail*"
									outline
									prepend-icon="email"
									required
								></v-text-field>
							</v-flex>
							<v-flex
								xs12
							>
								<v-text-field
									v-model="formData.password"
									:rules="rules.password"
									:type="showPassword ? 'text' : 'password'"
									:append-icon="showPassword ? 'visibility' : 'visibility_off'"
									prepend-icon="vpn_key"
									label="Senha*"
									outline
									@click:append="showPassword = !showPassword"
								/>
							</v-flex>
							<v-flex xs12>
								<span class="red--text caption">* campos obrigat√≥rios</span>
							</v-flex>
							<vue-recaptcha
								:sitekey="sitekey"
								ref="invisibleRecaptcha"
								size="invisible"
								badge="bottomleft"
								@verify="onVerifyRecaptcha"
								@expired="resetRecaptcha"
							/>
						</v-layout>
					</v-container>
				</v-form>
			</v-card-text>
			<v-card-actions>
				<v-container>
					<v-layout justify-space-between>
						<v-btn
							@click="handleForgotPassword"
							dark
							flat
							color="primary"
						>
							Resetar senha
						</v-btn>
						<v-btn
							@click="handleSubmit"
							color="primary"
							:disabled="disableSubmit"
						>
							<v-progress-circular
								v-if="disableSubmit"
								color="primary"
								indeterminate
							/>
							<span v-else>Login</span>
						</v-btn>
					</v-layout>
				</v-container>
			</v-card-actions>
		</v-card>
	</v-flex>
</template>

<script>
import VueRecaptcha from 'vue-recaptcha';
import validators from '../../../common/validators';
import { mapMutations } from 'vuex';

export default {
	components: {
		VueRecaptcha,
	},
	data() {
		return {
			showPassword: false,
			sitekey: process.env.MIX_SITE_KEY,
			valid: true,
			disableSubmit: false,
			formData: {
				email: '',
				password: '',
				recaptcha_response: null,
			},
			rules: {
				...validators,
			}
		}
	},

	methods: {
		...mapMutations('auth', ['setAuthenticated']),

		onVerifyRecaptcha(response) {
			this.formData.recaptcha_response = response;

			this.resetRecaptcha();

			axios.post(
				'/auth',
				this.formData,
			)
			.then(() => this.setAuthenticated(true))
			.catch((error) => {
				console.log(error.response);
			})
			.finally(() => {
				this.disableSubmit = false;
			});
		},

		resetRecaptcha() {
			this.$refs.invisibleRecaptcha.reset();
		},

		handleSubmit() {
			if (!this.disableSubmit && this.$refs.loginForm.validate()) {
				this.disableSubmit = true;
				this.$refs.invisibleRecaptcha.execute();
			}
		},

		handleForgotPassword() {
			this.$emit('toggleResetForm');
		}
	}
}
</script>

<style>

</style>
