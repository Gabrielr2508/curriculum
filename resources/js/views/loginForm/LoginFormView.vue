<template>
	<v-content class="grey lighten-2">
    	<v-container fluid fill-height>
        	<v-layout align-center justify-center>
				<login-form-card/>
        	</v-layout>
		</v-container>
	</v-content>
</template>

<script>
import LoginFormCard from './components/LoginFormCard';

export default {
	components: {
		LoginFormCard
	},
}
</script>

<style>

</style>
