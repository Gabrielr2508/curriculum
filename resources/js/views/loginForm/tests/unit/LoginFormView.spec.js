import { shallowMount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';

import LoginFormView from '@/views/loginForm/LoginFormView';
import LoginFormCard from '@/views/loginForm/components/LoginFormCard';

Vue.use(Vuetify);

describe('LoginFormView', () => {
    const build = () => {
        const wrapper = shallowMount(LoginFormView);

        return {
            wrapper,
            loginFormCard: () => wrapper.find(LoginFormCard),
        }
    }

    it('renders the component', () => {
        // arrange
        const { wrapper } = build();

        // assert
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders main child components', () => {
        // arrange
        const { loginFormCard } = build();

        // assert
        expect(loginFormCard().exists()).toBe(true);
    });
});
