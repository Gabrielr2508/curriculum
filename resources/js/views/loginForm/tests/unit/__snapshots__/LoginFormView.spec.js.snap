// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFormView renders the component 1`] = `
<v-content-stub tag="main" class="grey lighten-2">
  <v-container-stub tag="div" fluid="" fill-height="">
    <v-layout-stub tag="div" align-center="" justify-center="">
      <login-form-card-stub></login-form-card-stub>
    </v-layout-stub>
  </v-container-stub>
</v-content-stub>
`;
